boards/PDM/Inc/Io/Io_Efuse_22XS4200.h

```diff
@@ -0,0 +1,646 @@
+     LOW_CURRENT_THRESHOLD_SELECT_IOCL3 |                         \
+     HIGH_CURRENT_THRESHOLD_SELECT_IOCH1 | CURRENT_SENSE_RATIO_LOW)
+
+/* Serial Output Registers Enums */
+
+//! Enumeration for the status types
+typedef enum
+{
+    StatusType_NoFault = (0 << 0), //!< No fault
+    StatusType_OUT0    = (1 << 0), //!< Channel 0 Output state (ON/OFF)
+    StatusType_OUT1    = (1 << 1), //!< Channel 0 Output state (ON/OFF)
+    StatusType_FAULT0  = (1 << 2), //!< Fault on Channel 0
+    StatusType_FAULT1  = (1 << 3), //!< Fault on Channel 1
+    StatusType_R_FULL0 = (1 << 4), //!< Channel 0 Auto-retry Counter Full
+    StatusType_R_FULL1 = (1 << 5), //!< Channel 1 Auto-retry Counter Full
+    StatusType_POR     = (1 << 6), //!< Power-on Reset Has Occurred
+    StatusType_UV      = (1 << 7), //!< Under Voltage Fault
+    StatusType_OV      = (1 << 8)  //!< Over Voltage Fault
+} StatusType_e;
+
+//! Enumeration for the fault types
+typedef enum
+{
+    FaultType_NoFault = (0 << 0), //!< No fault
+    FaultType_OC      = (1 << 0), //!< Over Current fault
+    FaultType_SC      = (1 << 1), //!< Severe Short Circuit fault
+    FaultType_OT      = (1 << 2), //!< Over Temperature fault
+    FaultType_OS      = (1 << 3), //!< Output Shorted to Vpwr fault
+    FaultType_OLOFF   = (1 << 4), //!< Open Load in OFF State fault
+    FaultType_OLON    = (1 << 5), //!< Open Load in ON State fault
+    FaultType_OTW     = (1 << 8)  //!< Over Temperature Warning fault
+} FaultType_e;
+
+//! Enumeration for the PWM types
+typedef enum
+{
+    PWM_Type_DC_0 = (1 << 0), //!< PWM Duty Cycle bit 0
+    PWM_Type_DC_1 = (1 << 1), //!< PWM Duty Cycle bit 1
+    PWM_Type_DC_2 = (1 << 2), //!< PWM Duty Cycle bit 2
+    PWM_Type_DC_3 = (1 << 3), //!< PWM Duty Cycle bit 3
+    PWM_Type_DC_4 = (1 << 4), //!< PWM Duty Cycle bit 4
+    PWM_Type_DC_5 = (1 << 5), //!< PWM Duty Cycle bit 5
+    PWM_Type_DC_6 = (1 << 6), //!< PWM Duty Cycle bit 6
+    PWM_Type_DC_8 = (1 << 7), //!< PWM Duty Cycle bit 7
+    PWM_Type_ON   = (1 << 8)  //!< PWM ON
+} PWM_Type_e;
+
+//! Enumeration for the channel configurations types
+typedef enum
+{
+    ChannelConfigType_Delay_0              = (1 << 0), //!< Switch-on Delay bit 0
+    ChannelConfigType_Delay_1              = (1 << 1), //!< Switch-on Delay bit 1
+    ChannelConfigType_Delay_2              = (1 << 2), //!< Switch-on Delay bit 2
+    ChannelConfigType_SlewRate_0           = (1 << 3), //!< Slew Rate bit 3
+    ChannelConfigType_SlewRate_1           = (1 << 4), //!< Slew Rate bit 4
+    ChannelConfigType_DirecControl_Disable = (1 << 5), //!< Direct Control Disable
+    ChannelConfigType_OpenLoad_InOffState  = (1 << 6), //!< Open Load Detection in Off-state
+    ChannelConfigType_OpenLoad_InOnState   = (1 << 7), //!< Open Load Detection in On-state
+    ChannelConfigType_ShortCircuitDetection_Disable = (1 << 8) //!< Short-circuit to Vpwr Detection Disable
+} ChannelConfigType_e;
+
+struct E_Fuse
+{
+    bool         watch_dog_state;
```

Comment:
Since it's boolean, a better name could be `watchdog_is_on`? Or something like that

Reply:
hmm it's supposed to keep track of the value of this bit _**WDIN**_:
"Watchdog On" would correspond to the WatchDog_disable bit being set to 1:
![image](https://user-images.githubusercontent.com/25499626/81451106-9a3b6000-9138-11ea-8ce3-4b53efedf371.png)
Toggling of the WDIN is the method of kicking/feeding the watchdog, it's not what indicates if it's active of not.

Reply:
We should be adding this information as a comment in the struct member then, the name is a bit confusing

Reply:
something like `wdin_bit_to_set` plus a descriptive comment like i mentioned in the above comment should do